### DISCUSSION TITLE
Serverless, or semi serverless setup

### DISCUSSION THREAD

**Discussion by Aevrin** [FreshRSS/FreshRSS#4494] [Ideas]
I'm not a programmer at all, and I've been using this for several years.  I was playing around on heroku and got it deployed using the Heroku postgres setup on the hobby tier.  I had to make a few changes to the composer.json/composer.lock that maybe could have been avoided with ext-mbstring and ignore local, but there may be a more elegant way to do this,

Anyway, the long and short of the idea would be rolling all of this into one composer that's already good to go, (maybe include addons as well) and have a single one-shot deployment for anyone who wants to use it to make it way easier to use.

**Comment by Alkarex** [comment]
Hello,
Feel free to post your changes, so we can potentially include them.

Edit: I have made a quick patch https://github.com/FreshRSS/FreshRSS/pull/4497 Could you give it a try?

For inspiration, we have a few container options already, including [Docker](https://github.com/FreshRSS/FreshRSS/tree/edge/Docker) and a few other https://github.com/FreshRSS/FreshRSS#automated-install

And our CLI allows fully automated install (see e.g. our docker-compose examples).

**Reply by Aevrin** [reply]
Oops, didn't reply, posted a new comment.

**Comment by Aevrin** [comment]
Finally got to test out a fresh deployment, looks pretty nice.  I was able to simply clone the current repo, push it to heroku, and not just did it deploy without issue this time, I was also able to select SQLITE which worked fine.  I guess to build on this current discussion, I'm not sure how to install extensions with this setup.
Would there be a way an admin could have a 1 click deployment for the supported addin's?  For my primary deployment, I simply use a post bash install script that grabs the extensions I need, but an in app deployment might be cool (or a modifiable deployment package) could be a fun enhancement.  Either way, thanks for the changes.
